src/core/document.js

```diff
@@ -482,7 +479,9 @@ class Page {
     return this._parsedAnnotations.then(function (annotations) {
       const annotationsData = [];
       for (let i = 0, ii = annotations.length; i < ii; i++) {
-        if (!intent || isAnnotationRenderable(annotations[i], intent)) {
+        // Get the annotation even if it's hidden because
+        // JS can change its display.
+        if (annotations[i].viewable) {
```

Comment:
You probably want to keep `!intent || ...` here, since otherwise it'd be an (indirect) API change as.